### DISCUSSION TITLE
Delete all tuples related to a deleted object

### DISCUSSION THREAD

**Discussion by leah-mq** [openfga/community#219] [Q&A] [answered]
Hello,

I would like to base my authorization rules on openfga for my .net app. When an object is deleted from the database, I would like to delete the the tuples where this object is involved.

Is there a way to delete all existing tuples for an object when this one is deleted from database? 
Something like  the following code, deleting everything related to document:Z :
` var options = new ClientListObjectsOptions {
    AuthorizationModelId = "1uHxCSuTP0VKPYSnkq1pbb1jeZw",
};

var body = new ClientWriteRequest() {
    Deletes = new List<ClientTupleKey>() {
    new() { User = "user:\*", Relation = "*", Object = "document:Z" }
  },
};
var response = await fgaClient.Write(body, options);`

**Comment by aaguiarz** [comment, marked as answer]
Hi @leah-mq 

It's currently not possible, it's in our roadmap here https://github.com/openfga/roadmap/issues/34

You'd need to read them first, and then delete them :(

**Reply by leah-mq** [reply]
Hi @aaguiarz

Thank you
